### ISSUE TITLE
Resolve fails silently if passed function that takes arguments.

### ISSUE THREAD

**Issue by TimeBomb** [angular-ui/ui-router#2803]
When using the state provider to create a state that has a templateProvider, controllerProvider, and a function to resolve, if that function to be resolved takes arguments, then it does not appear to be invoked.

Example that doesn't resolve the function (`doStuff` in this case):
```
function doStuff(optionalArg) {
	debugger;
	// returns promise here
}

$stateProvider.state('things', {
	url: '/things',
	templateProvider: ['stuff', function(stuff) {
		return stuff.template;
	}],
	controllerProvider: ['stuff', function(stuff) {
		return stuff.controller;
	}],
	resolve: {
		stuff: doStuff
	}
});
```
In the above example, `debugger;` is never hit.

Whereas if we remove the argument that the `doStuff` function takes, the debugger is hit:
```
function doStuff() {
	debugger;
	// returns promise here
}

$stateProvider.state('things', {
	url: '/things',
	templateProvider: ['stuff', function(stuff) {
		return stuff.template;
	}],
	controllerProvider: ['stuff', function(stuff) {
		return stuff.controller;
	}],
	resolve: {
		stuff: doStuff
	}
});
```

I figured this out through trial and error. It makes enough sense that it doesn't try to resolve it, but logging an error to the console when you try to resolve a function that has arguments would be very helpful, and would have saved me time in this scenario. I'm opening this issue to suggest that that is done.

For anyone looking for a workaround to this issue, you can resolve it successfully like this:
```resolve: {
    stuff: function() {
        doStuff();
}```

Or if you want to use the stateParams or another provider to get arguments to pass to your function:
```resolve: {
		stuff: ['$stateParams', function($stateParams) {
			doStuff($stateParams);
		}]
	}```

**Comment by christopherthielen** [comment]
1.0 now includes a [defaultErrorHandler](https://ui-router.github.io/docs/latest/classes/state.stateservice.html#defaulterrorhandler) which logs transition failures to the console.  The handler can be overridden with custom logic.

http://plnkr.co/edit/lc8HJTr4JUpOgwnAmXN0?p=preview

Closing as fixed in 1.0.

**Comment by christopherthielen** [comment]
Note: in legacy versions, use 
```
$rootScope.$on('$stateChangeError', function(event) {
//logic
});
```
